tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/AdminTests.cs

```diff
@@ -29,6 +31,123 @@ public class AdminTests
+            RecordingHandler testRecordingHandler = new RecordingHandler(Directory.GetCurrentDirectory());
+            var httpContext = new DefaultHttpContext();
+            
+            string requestBody = @"[
+    {
+        ""Name"": ""GeneralRegexSanitizer"",
+        ""Body"": { 
+            ""regex"": ""[a-zA-Z]?"",
+            ""value"": ""hello_there"",
+            ""condition"": {
+                ""UriRegex"": "".+/Tables""
+            }
+        }
+    },
+    {
+        ""Name"": ""HeaderRegexSanitizer"",
+        ""Body"": { 
+            ""key"": ""Location"",
+            ""value"": ""https://fakeazsdktestaccount.table.core.windows.net/Tables""
+        }
+    }
+]";
+
+            httpContext.Request.Body = TestHelpers.GenerateStreamRequestBody(requestBody);
+            httpContext.Request.ContentLength = httpContext.Request.Body.Length;
+            testRecordingHandler.Sanitizers.Clear();
+
+            var controller = new Admin(testRecordingHandler, _nullLogger)
+            {
+                ControllerContext = new ControllerContext()
+                {
+                    HttpContext = httpContext
+                }
+            };
+            await controller.AddSanitizers();
+
+
+            Assert.Equal(2, testRecordingHandler.Sanitizers.Count);
+
+            Assert.True(testRecordingHandler.Sanitizers[0] is GeneralRegexSanitizer);
+            Assert.True(testRecordingHandler.Sanitizers[1] is HeaderRegexSanitizer);
+        }
+
+        [Fact]
+        public async void TestAddSanitizersThrowsOnSingleBadInput()
+        {
+            RecordingHandler testRecordingHandler = new RecordingHandler(Directory.GetCurrentDirectory());
+            var httpContext = new DefaultHttpContext();
+            
+            string requestBody = @"[
+    {
+        ""Name"": ""GeneralRegexSanitizer"",
+        ""Body"": { 
+            ""regex"": ""[a-zA-Z]?"",
+            ""value"": ""hello_there"",
+            ""condition"": {
+                ""UriRegex"": "".+/Tables""
+            }
+        }
+    },
+    {
+        ""Name"": ""BadRegexIdentifier"",
+        ""Body"": { 
+            ""key"": ""Location"",
+            ""value"": ""https://fakeazsdktestaccount.table.core.windows.net/Tables""
+        }
+    }
+]";
+
+            httpContext.Request.Body = TestHelpers.GenerateStreamRequestBody(requestBody);
+            httpContext.Request.ContentLength = httpContext.Request.Body.Length;
+            testRecordingHandler.Sanitizers.Clear();
+
+            var controller = new Admin(testRecordingHandler, _nullLogger)
+            {
+                ControllerContext = new ControllerContext()
+                {
+                    HttpContext = httpContext
+                }
+            };
+            var assertion = await Assert.ThrowsAsync<HttpException>(
+                async () => await controller.AddSanitizers()
+            );
+
+            assertion.StatusCode.Equals(HttpStatusCode.BadRequest);
```

Comment:
One could also assert that no sanitizers were added.

Reply:
Absolutely. But we're also verifying the exception that is being thrown is the correct one 👍